File: 1850/E.cpp
#include <bits/stdc++.h>
using namespace std;

#define ll unsigned long long

void solve()
{
	ll n, c;
	cin >> n >> c;
	ll a[n];
	for (ll &x : a)
		cin >> x;
	ll sum_c = 0, sum_b = 0, sum_a = 0;
	for (ll x : a)
		sum_c += x * x, sum_b += x * 4, sum_a += 4;
	sum_c -= c;
	
}

int main()
{
	cin.tie(0) -> sync_with_stdio(0);
	int t;
	cin >> t;
	while (t--)
		solve();
	return 0;
}

File: 1850/E.py
from math import sqrt


def solve():
	n, c = [int(x) for x in input().split()]

	arr = list(map(int, input().split()))

	sum_a = 0
	sum_b = 0
	sum_c = 0
	for i in arr:
		sum_a += 4
		sum_b += i * 4
		sum_c += i * i
	
	sum_c -= c
	print(round((-sum_b + sqrt(sum_b * sum_b - 4 * sum_a * sum_c)) / (2 * sum_a)))


t = int(input())

for i in range(t):
	solve()